### ISSUE TITLE
[tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: undefined is not an object (evaluating 'a.View.propTypes.style')

### ISSUE THREAD

**Issue by NazarK** [magus/react-native-facebook-login#294]
this line caused my app to crash (strangely worked fine in debug build and crashed in release), seems they renamed View.propTypes.style (as temporary fix I just removed this line in my ./node_modules/)

https://github.com/magus/react-native-facebook-login/blob/4a2e667a016a9196311a3dde597cc1b46112fd74/index.ios.js#L93

**Comment by kayzenkayzen** [comment]
Some issue.

How to fix it without node modules changes?

**Comment by aramkarapetian** [comment]
Fork and set source to your repo instead.

**Comment by NazarK** [comment]
Yes, I forked for myself and edited package.json, like this
 "dependencies": {

    "react-native-facebook-login": "git+https://git@github.com/NazarK/react-native-facebook-login.git",

  }

**Comment by catchkuo** [comment]
I use simulator was fine, but build to IOS device got same issue.

System group container for systemgroup.com.apple.configurationprofiles path is /private/var/containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2018-02-02 17:14:25.159121+0800 YesEvent[5854:2890540] [MC] Reading from public effective user settings.
2018-02-02 17:14:25.164172+0800 YesEvent[5854:2890640] Missing server connection for kCTConnectionInvalidatedNotification
2018-02-02 17:14:25.252 [error][tid:com.facebook.react.JavaScript] undefined is not an object (evaluating 'u.View.propTypes.style')
2018-02-02 17:14:25.253616+0800 YesEvent[5854:2890650] undefined is not an object (evaluating 'u.View.propTypes.style')
2018-02-02 17:14:25.256 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: undefined is not an object (evaluating 'u.View.propTypes.style')
2018-02-02 17:14:25.256173+0800 YesEvent[5854:2890639] Unhandled JS Exception: undefined is not an object (evaluating 'u.View.propTypes.style')
2018-02-02 17:14:25.260825+0800 YesEvent[5854:2890639] *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: undefined is not an object (evaluating 'u.View.propTypes.style')', reason: 'Unhandled JS Exception: undefined is not an object (evaluating 'u.View.prop..., stack:

**Comment by NazarK** [comment]
this error appeared for me only when I was testing "Release" build on simulator

**Comment by acro5piano** [comment]
I got the same error. Finally I run this script after installation (or before Release build):

```js
# Update react-native-facebook-login
# refs:
#     - https://github.com/react-navigation/react-navigation/issues/1352
#     - https://github.com/magus/react-native-facebook-login/issues/294
perl -i -pe 's/View,$/View, ViewPropTypes,/' ./node_modules/react-native-facebook-login/index.ios.js
perl -i -pe 's/View.propTypes.style,$/ViewPropTypes,/' ./node_modules/react-native-facebook-login/index.ios.js
```

This is undesirable Hack but temporarily solution here.

**Comment by magus** [comment]
Thanks for reporting this, the fix is out in version 1.6.1!

**Comment by Soren-Knudsen** [comment]
I'm using 1.6.1 and the issue `[error][tid:com.facebook.react.JavaScript] undefined is not an object (evaluating 'd.View.propTypes.style')` is still present.

**Comment by acro5piano** [comment]
@Soren-Knudsen I had the same issue but after I upgrade to 1.6.1 the error disappeared. 
Did you run the following commands after upgrading?

```
rm -rf node_modules
yarn install # or npm install
yarn start --reset-cache
```

**Comment by Soren-Knudsen** [comment]
@acro5piano: Thanks for a quick response. I haven't tried `yarn start --reset-cache`. I'll give that a go.

**Comment by Soren-Knudsen** [comment]
@acro5piano : Tried `yarn cache clean && rm -fr node_modules/ && yarn && yarn start --reset-cache` without luck. Still getting the error when building my project in Xcode in Release mode :(

**Comment by acro5piano** [comment]
@Soren-Knudsen Did you succeed when building in Debug mode?

**Comment by Soren-Knudsen** [comment]
@acro5piano : Yes without problems. Somehow the issue seems to persist when building it in release mode even though I cleaned out the yarn cache and reinstalled all my node modules.

**Comment by acro5piano** [comment]
@Soren-Knudsen Hmmm this is strange. 
Xcode cache might remain so `⌘ + k` and `⌘ + ⇧ + k` and `rm -rf Library/Developer/Xcode/DerivedData` can help.
Also if you succeed when building Debug mode, then your problem sounds nothing to do with `react-native-facebook-login`.

**Comment by Soren-Knudsen** [comment]
@acro5piano: I'm able to build it in both in `Debug` and `Release` mode, but it throws an unhandled exception in `Release` mode when the app starts (See snippet below).

```
2018-04-01 17:55:17.347 [error][tid:com.facebook.react.JavaScript] undefined is not an object (evaluating 'd.View.propTypes.style')
2018-04-01 17:55:17.349765+0200 XXXX[6869:30684] undefined is not an object (evaluating 'd.View.propTypes.style')
2018-04-01 17:55:17.355 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: undefined is not an object (evaluating 'd.View.propTypes.style')
2018-04-01 17:55:17.354978+0200 XXXX[6869:30665] Unhandled JS Exception: undefined is not an object (evaluating 'd.View.propTypes.style')
2018-04-01 17:55:17.356 [error][tid:com.facebook.react.JavaScript] Module AppRegistry is not a registered callable module (calling runApplication)
```

**Tried the following:**

- `yarn cache clean && rm -fr node_modules/ && yarn && yarn start --reset-cache`
- Xcode build clean
- `rm -rf Library/Developer/Xcode/DerivedData`
- Rebooted my computer and did everything over again.

I also tried using `git+https://git@github.com/NazarK/react-native-facebook-login.git` without luck. Do you have any other suggestions I could try?

**Comment by acro5piano** [comment]
@Soren-Knudsen Thanks for the detail information. I saw the error when I used `1.6.0`.

At first, please ensure you successfully installed and using `1.6.1`. If you run 

```
grep viewPropTypes -r node_modules/react-native-facebook-login/
```

in your project root and the output is something like

```
node_modules/react-native-facebook-login//example/components/facebook/FBLoginMock.js:const viewPropTypes = ViewPropTypes || View.propTypes;
node_modules/react-native-facebook-login//example/components/facebook/FBLoginMock.js:    style: viewPropTypes.style,
node_modules/react-native-facebook-login//index.ios.js:const viewPropTypes = ViewPropTypes || View.propTypes;
node_modules/react-native-facebook-login//index.ios.js:  style: viewPropTypes.style,
```

then it's ok.

Second, which is your RN version? `View.propTypes` deprecated after `v44` so something bad might happen before `v44` (though I couldn't find anything wrong within the code).

refs: 

- https://github.com/facebook/react-native/issues/16542
- https://qiita.com/devneko/items/9f356f610099fc254704 (ja)

Third, if the error cannot be delete...
Were I you I would create a script that delete the following line in `node_modules/react-native-facebook-login/index.ios.js` with perl or something, and run the script before Release build as just for temporally solution.

```js
style: viewPropTypes.style,
```

Or you can use this library,  I just found.

https://github.com/reactjs/react-codemod#reactnative-view-proptypes

**Comment by Soren-Knudsen** [comment]
**Steps**

1. Output from: `grep viewPropTypes -r node_modules/react-native-facebook-login/`

```
node_modules/react-native-facebook-login//example/components/facebook/FBLoginMock.js:const viewPropTypes = ViewPropTypes || View.propTypes;
node_modules/react-native-facebook-login//example/components/facebook/FBLoginMock.js:    style: viewPropTypes.style,
node_modules/react-native-facebook-login//index.ios.js:const viewPropTypes = ViewPropTypes || View.propTypes;
node_modules/react-native-facebook-login//index.ios.js:  style: viewPropTypes.style,
```

2. Which is your RN version? 
   - node: 9.9.0
   - react-native: 0.54.4
   - react-native-facebook-login: 1.6.1

@acro5piano: Are you able to build and run your application in release mode when using `react-native` ^0.54.4 with `react-native-facebook-login` ^1.6.1? Or are you using an older version of `react-native`?

**Comment by acro5piano** [comment]
@Soren-Knudsen My environment is:

- node: 8.4.0
- react-native: 0.52.0
- react-native-facebook-login: ^1.6.1

Little bit older than your version of `react-native`. 

I am able to build and run my application both in Debug mode and Release mode.
